tests/integration.rs

```diff
@@ -362,6 +362,42 @@ fn test_version_sort_overwrite_by_sizesort() {
         .stdout(predicate::str::is_match("11\n2\n$").unwrap());
 }
 
+#[test]
+#[cfg(target_os = "linux")]
+fn test_bad_utf_8_extension() {
+    let tmp = tempdir();
+    Command::new("python3")
+        .arg("-c")
+        .arg(format!(
+            r#"import pathlib; pathlib.Path('{}/bad.extension\udca7\udcfd').touch()"#,
+            tmp.path().to_str().unwrap()
+        ))
+        .output()
+        .expect("failed to create file");
```

Comment:
try with double slashes
```suggestion
    Command::new("sh")
        .arg("-c")
        .arg("touch $(printf 'bad.extension\\xdca7\\xdcfd')")
        .output()
        .unwrap();
```

Reply:
have your try to run the code?

double backslashes fix the rust error report, but created a file named `'bad.extension\xdca7\xdcfd'` not the utf 8 code.

I have done several tries before using python

Reply:
Yeah if i did a project with main.rs:
```rust
use std::process::Command;

fn main() {
    Command::new("sh")
        .arg("-c")
        .arg("touch $(printf 'bad.extension\\xdca7\\xdcfd')")
        .output()
        .unwrap();
}
```
and when i run `ls` i get `'bad.extension'$'\334''a7'$'\334''fd'`.

Reply:
that's really weird, I can't reproduce your result
![image](https://user-images.githubusercontent.com/3764335/95417037-a422f280-0966-11eb-9160-c7b97124a303.png)

Reply:
@zwpaper Are you running from within a linux docker image on a mac?

Is it possible that that could affect the result?

Reply:
that's also my concern, so I was setting up a VM in fusion, big sur broke my VirtualBox

Reply:
the same result on debian 10 in fusion vm.

![image](https://user-images.githubusercontent.com/3764335/95444071-f62c3e00-098f-11eb-8a5a-31b2f2a26268.png)

@0jdxt could you help me to reproduce this?

Reply:
I'll submit a PR just to see if it works with the CI.